frontend/src/pages/offers/PalletsEditor.tsx

```diff
@@ -0,0 +1,131 @@
+      return
+    }
+
+    createLineItem({
+      variables: { palletId },
+      refetchQueries: [
+        {
+          query: PalletDocument,
+          variables: { id: selectedPalletId },
+        },
+      ],
+    }).then((data) => {
+      setSelectedLineItemId(data.data?.addLineItem.id)
+    })
+  }
+
+  return (
+    <section className="flex border border-gray-100">
+      <div className="w-1/3">
+        <ul className="divide-y divide-gray-100">
+          {pallets.map((pallet) => (
+            <li
+              key={pallet.id}
+              className="bg-white"
+              onClick={() => selectPallet(pallet.id)}
+            >
+              <div
+                className={cx('p-4 bg-white border-l-4 border-transparent', {
+                  ' border-blue-500': selectedPalletId === pallet.id,
+                })}
+              >
+                <div className="mb-2 font-semibold">Pallet {pallet.id}</div>
+                {selectedPalletId === pallet.id && (
+                  <div className="border border-gray-200 flex flex-col mt-2 rounded">
+                    {selectedPalletData.data?.pallet.lineItems.map((item) => (
+                      <button
+                        type="button"
+                        className={cx('px-4 py-2 text-left border-l-4', {
+                          'border-blue-500': item.id === selectedLineItemId,
+                        })}
+                        key={item.id}
+                        onClick={(e) => {
+                          e.stopPropagation()
+                          setSelectedLineItemId(item.id)
+                        }}
+                      >
+                        Item {item.id}
+                      </button>
+                    ))}
+                    <button
+                      type="button"
+                      className="px-4 py-2 text-left border-t"
+                      onClick={(e) => {
+                        e.stopPropagation()
+                        addLineItem(pallet.id)
+                      }}
+                    >
+                      Add a line item
+                    </button>
+                  </div>
+                )}
+              </div>
+            </li>
+          ))}
+        </ul>
+      </div>
+      <div className="flex-grow border-l border-gray-100">
+        {selectedPalletId == null && (
+          <div className="flex h-full w-full items-center justify-center bg-gray-50 text-gray-500">
+            <p>← Select a pallet</p>
+          </div>
+        )}
+        {selectedPalletId != null && selectedLineItemId == null && (
+          <div className="flex h-full w-full items-center justify-center bg-gray-50 text-gray-500">
+            <p>← Select a line item</p>
+          </div>
+        )}
+        {selectedPalletId != null && selectedLineItemId != null && (
+          <div className="h-full w-full p-4">
+            <LineItemForm lineItemId={selectedLineItemId} />
+          </div>
+        )}
+      </div>
```

Comment:
WIP code. Will break it up into components once the features have been finalized.